### PR TITLE
Migrate Roslyn Insertion Tool

### PR DESCRIPTION
**Pull request by tmeschter**
Move the Roslyn Insertion Tool to the roslyn-tools repository and integrate it into the build.

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs]: Missing copyright

File: src/RoslynInsertionTool/RoslynInsertionTool/BuildVersion.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool/BuildVersion.cs]: Missing copyright

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
+﻿using System;
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs]: Missing copyright

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs]: Missing copyright.

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Roslyn.Insertion
+{
+    static partial class RoslynInsertionTool
+    {
+        private static async Task<bool> CanBuildPartitionAsync(string relativePathToPartition, CancellationToken cancellationToken)
+        {
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs]: Do we need this file anymore? it mentions Razzle build and didn't thin kwe did that for insertion.

File: src/RoslynInsertionTool/RoslynInsertionTool/VersionsUpdater.cs [hunk 6]
```diff
@@ -0,0 +1,218 @@
+﻿using System;
```
Comment (jaredpar) [inline src/RoslynInsertionTool/RoslynInsertionTool/VersionsUpdater.cs]: Missing copyright.

File: src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs (hunk 1 above)
Comment (tmeschter) [reply src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs]: Sigh. I guess somehow we never ended up with an analyzer for copyright headers?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs (hunk 5 above)
Comment (tmeschter) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs]: @jaredpar I'm not sure. There are a fair number of options for RIT.exe, and no doubt some of them are outdated/broken/irrelevant now. However, I'd rather not block getting this into roslyn-tools on cleaning that up.

Comment (jaredpar) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.RazzleBuild.cs]: Agree. This isn't blocking feedback. More curiousity.

File: src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs (hunk 1 above)
Comment (jmarolf) [reply src/RoslynInsertionTool/RoslynInsertionTool.Commandline/RoslynInsertionToolCommandline.ShowHelp.cs]: @dpoeschl

**Pull request merged**